### ISSUE TITLE
Update pyoculus on pypi

### ISSUE THREAD

**Issue by abaillod** [zhisong/pyoculus#4]
Hi Zhisong,

Can we update the pyoculus on pypi with the latest version? A student (Rohan Lopez) here at Columbia would need to use some of the features of version 0.3.0 (mainly the `SimsgeoBiotSavart` class).

He is using a docker image of simsopt. We tried installing pyoculus 0.3.0 by pulling the git repository and doing `pip install .` but it fails. It would be more convenient to go through `pip install`!

**Comment by zhisong** [comment]
In principle no problem, since I have not touched the things being used in simsopt. But who knows, something may break! To make sure, the best way is to test simsopt with the latest version. If things work, we can update pypi.

Note that the new versions of numpy package (or setuptools?) may cause the installation scripts in setup.py to fail. (Not just pyoculus, but may other codes too. I think they changed their interface for some reason.) Two options to proceed, either we change setup.py to adapt to the new version, or we use an older version of numpy or setuptools (say, some versions dated back to 2021).

**Comment by abaillod** [comment]
That's the thing - I'm facing some compilation issues with SPEC and I want to use the simsopt docker image instead of going through the local installation. Therefore I can't test if the latest version of pyoculus breaks something in simsopt..,.

Regarding numpy and setuptools I guess the best to proceed is to adapt the script so that it works with the latest version.

**Comment by abaillod** [comment]
According to https://stackoverflow.com/questions/70583969/pyinstaller-no-module-named-numpy-distutils, we need to use `numpy <= 1.21.5` to make it work. This requires to use `python<=3.10`.

**Comment by zhisong** [comment]
Do you know what is the requirement of SIMSOPT?

**Comment by zhisong** [comment]
Yes, I tested the latest python (3.11) and numpy (1.26). No chance... My worry is that if we update to the new numpy, the old versions (e.g. in SIMSOPT docker) will break...

**Comment by abaillod** [comment]
Maybe the easiest is then to enforce the use of `numpy<=1.25`. Looking at simsopt, they require `numpy>=1.19.4`. Currently, the simsopt docker image contains `numpy==1.26.0`, but I guess it could be recompiled with a downgraded version of `numpy` that satisfies our constraints

**Comment by smiet** [comment]
Hi everyone, 

I will get on fixing the install script, but I do not really see the issue. 

@abaillod: the docker container is failing because it does not have GCC. The following one-liner will install the latest version of pyoculus in your container: 
```bash
$  docker run -it --rm hiddensymmetries/simsopt
# yes | apt-get install gfortran && pip install git+https://github.com/zhisong/pyoculus.git
```

@zhisong: Numpy is backwards compatible, and if you want to not faff around too much, put "oldest-supported-numpy" in the `[build-system]` `requires = []` field. 
The build will be performed using the oldest supported numpy for your python-version and platform (and not whine about the change in `distutils`), but upon execution python is still linking against the system's numpy. 
(A trick I stumbled upon when I was getting very frustrated with the ongoing f90wrap issues). 

I will make a pull request for the change tomorrow as I need to sprint to daycare, maybe get around to streamlining the build later.

**Comment by zhisong** [comment]
Thanks both. I will email you a script to upload to Pypi. Once you are sure it will work with simsopt...

**Comment by smiet** [comment]
Uploaded branch fix_setuptools, and spent all day fighting with simsopts' unit tests because I cant get the old version to run cleanly. 

Anyways, @abaillod can you checkout this branch and see if it installs correctly on your system(s)? 
`python -c"import pyoculus; print(pyoculus.__version__)"` should give 0.3.1 if all is good. 

If all is indeed good, I will go ahead and push the new to pypi, then ask simsopt to bake a new docker container, and life should be a bit better for all of us.

**Comment by smiet** [comment]
@abaillod also don't forget to clear all build files with a 'git clean -xfd'

**Comment by zhisong** [comment]
I tested a fresh install on a fresh conda environment with the latest python and numpy. Seems fine. I don't have a simsopt container so no idea if that will work.

**Comment by smiet** [comment]
Closing as issue is addressed in PR #6
